lib/salestation/web/ip_address.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Salestation
+  class Web < Module
+    module IPAddress
+        include Deterministic
+
+        def self.extract(request)
+          ip_address = if request['HTTP_X_FORWARDED_FOR'].nil?
+             request['REMOTE_ADDR']
+            else
+             request['HTTP_X_FORWARDED_FOR'].split(',').first
+            end
+          Result::Success(ip_address)
+        end
+    end
+  end
+end
```

Comment:
you had nice tests in the other project that should also be brought here

Reply:
If this approach is correct - yes, I will add the tests :)